extensions/keycloak-authorization/runtime/src/main/java/io/quarkus/keycloak/pep/KeycloakPolicyEnforcerAuthorizer.java

```diff
@@ -0,0 +1,191 @@
+package io.quarkus.keycloak.pep;
+
+import java.security.Permission;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.inject.Singleton;
+
+import org.keycloak.AuthorizationContext;
+import org.keycloak.adapters.KeycloakDeploymentBuilder;
+import org.keycloak.adapters.authorization.KeycloakAdapterPolicyEnforcer;
+import org.keycloak.adapters.authorization.PolicyEnforcer;
+import org.keycloak.representations.adapters.config.AdapterConfig;
+import org.keycloak.representations.adapters.config.PolicyEnforcerConfig;
+
+import io.quarkus.arc.AlternativePriority;
+import io.quarkus.oidc.OidcConfig;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.runtime.QuarkusSecurityIdentity;
+import io.quarkus.vertx.http.runtime.security.HttpAuthorizer;
+import io.quarkus.vertx.http.runtime.security.QuarkusHttpUser;
+import io.vertx.ext.web.RoutingContext;
+
+@Singleton
+@AlternativePriority(1)
+public class KeycloakPolicyEnforcerAuthorizer extends HttpAuthorizer {
```

Comment:
So this is not really integrating with our HTTP permission layer, but just completely replacing it?

We need to discuss this, as it means that all other security rules will be ignored.

Reply:
Actually I think this needs two layers, the existing policy based path stuff, and the ability to add additional checks that do their own path matching (which servlet web.xml integration will need).

Reply:
@stuartwdouglas this extension is kind of a replacement of Quarkus permissions. Differently than what we have in Quarkus, the policies are externalized, completely decoupled from the application, and centralized managed through Keycloak.

The functionality provided by our policy enforcer has more requirements in regards to path mapping and matching. We also have some requirements in regards to not authorized or forbidden responses (e.g.: when doing UMA).

This extension is all about bringing a Keycloak specific feature into Quarkus. We have users familiar with the configuration I'm proposing here, making easier to migrate their applications.

I'm not sure if it would make sense to have part of your resources protected by Quarkus policies and other parts protected by Keycloak. However, we could review the changes I'm proposing and delegate to the Quarkus permission layer when the extension denies access to a resource.